pkg/job_runner/awx_job_runner/awx_job_runner.go

```diff
@@ -14,34 +14,57 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package awx_job_runner
 
 import (
 	"fmt"
 
 	"github.com/golang/glog"
+	"golang.org/x/sync/syncmap"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/moolitayer/awx-client-go/awx"
 	"github.com/openshift/autoheal/pkg/alertmanager"
 	"github.com/openshift/autoheal/pkg/apis/autoheal"
+
+	"github.com/openshift/autoheal/pkg/config"
+	"github.com/openshift/autoheal/pkg/metrics_exporter"
 )
 
-func (h *Healer) runAWXJob(rule *autoheal.HealingRule, action *autoheal.AWXJobAction, alert *alertmanager.Alert) error {
-	var err error
+type AWXJobRunner struct {
+	config *config.AWXConfig
+
+	activeJobs *syncmap.Map
+}
+
+func NewAWXJobRunner() *AWXJobRunner {
+	awxJubRunner := new(AWXJobRunner)
+	return awxJubRunner
+}
 
+func (r *AWXJobRunner) Init(config *config.Config, stopCh <-chan struct{}) *AWXJobRunner {
+	r.config = config.AWX()
+	r.activeJobs = new(syncmap.Map)
+	go wait.Until(r.runActiveJobsWorker, r.AWX().JobStatusCheckInterval(), stopCh)
+	return r
+}
+
+func (r *AWXJobRunner) RunJob(rule *autoheal.HealingRule, action interface{}, alert *alertmanager.Alert) error {
+	var err error
+	awxAction := action.(*autoheal.AWXJobAction)
 	// Get the AWX connection details from the configuration:
-	awxAddress := h.config.AWX().Address()
-	awxProxy := h.config.AWX().Proxy()
-	awxUser := h.config.AWX().User()
-	awxPassword := h.config.AWX().Password()
-	awxCA := h.config.AWX().CA()
-	awxInsecure := h.config.AWX().Insecure()
+	awxAddress := r.AWX().Address()
```

Comment:
I think that this `AWX` method unnecessary hides the fact that we are accessing the configuration. Please avoid it.